### PR TITLE
feat(forms): add Form.Isolation to isolate data context temporarily

### PR DESCRIPTION
**Pull request by tujoworker**
When it comes to copy the isolated data over to the main context, we need a term. Question; is the term "dispatch" good to use? I also think "commit" could work. Its a "one way transfer", so I don't want to use the term "sync". 

Any thoughts?

Also, I only have one example for now. And its a pure technical example. 
Maybe @andlbrei has a more practical example?

Here are the docs: https://eufemia-git-feat-isolate-eufemia.vercel.app/uilib/extensions/forms/Form/Isolation/

### CODE REVIEW

**Comment by andlbrei** [comment]
I like both alternatives. Commit feels more right to me, but that might just be redux trauma affecting my opinion. 😅 

This design shows one example where we will use this.
bilde

We have a Hovedkontaktperson.
We can edit this person, in which case we only allow editing of phone number and email address.
We can add a new Hovedkontaktperson by choosing from a list of persons who already have roles in the company.
Or we can add a completely new person, which the design above shows, in which case we need to fill in more information (date of birth, name and address).

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+showTabs: true
+hideInMenu: true
+---
+
+## Description
+
+`Form.Isolation` lets you isolate parts of your from so data and validations are not shared between the `Form.Handler` until you want to.
+
+Its a provider that lets you provide a `schema` or `data` very similrar to what the [Form.Handler](/uilib/extensions/forms/Form/Handler/) component does.
```
Comment (andlbrei) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx:10]:
```suggestion
Its a provider that lets you provide a `schema` or `data` very similar to what the [Form.Handler](/uilib/extensions/forms/Form/Handler/) component does.
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+### Good to know
+
+- It needs to be used inside of a `Form.Handler` component.
+- All fields inside needs to validate sucssfully before the isolated data can be dispatched, like the `Form.Handler` does before it submits.
```
Comment (andlbrei) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx:15]:
```suggestion
- All fields inside needs to validate successfully before the isolated data can be dispatched, like the `Form.Handler` does before it submits.
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+- Input fields are prevented from submitting the form when pressing enter. Pressing enter on input fields will dispatch the isolated data to the `Form.Handler` context instead.
+- You can provide a `schema`, `data` or `defaultData` like you would do with the `Form.Handler`.
+- You can also provide `data` or `defaultData` to the `Form.Handler`, definding the data that will be used for the isolated data.
```
Comment (andlbrei) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx:18]:
```suggestion
- You can also provide `data` or `defaultData` to the `Form.Handler`, defining the data that will be used for the isolated data.
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+- `onChange` on the `Form.Handler` will be called when the isolated gets dispatched.
```
Comment (andlbrei) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx:19]:
```suggestion
- `onChange` on the `Form.Handler` will be called when the isolated data gets dispatched.
```
Not sure about this one, but it should be improved

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx [hunk 5]
```diff
@@ -0,0 +1,92 @@
+---
+showTabs: true
+hideInMenu: true
+---
+
+## Description
+
+`Form.Isolation` lets you isolate parts of your from so data and validations are not shared between the `Form.Handler` until you want to.
```
Comment (joakbjerk) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Isolation/info.mdx:8]:
```suggestion
`Form.Isolation` lets you isolate parts of your form so data and validations are not shared between the `Form.Handler` until you want to.
```

**Review by joakbjerk — APPROVED** [review]

**Pull request merged**